ft(API): User can delete a booking

amkayondo opened the pull request:
#### What does this PR do?
Adds files that enable User can delete a booking
#### Description of Task to be completed?
Users can delete their booking.
#### How should this be manually tested?
send `DELETE` - `/bookings/<:bookingId>`
#### Any background context you want to provide?
`200` - OK
`400` - BAD REQUEST
`404` - NOT FOUND
#### What are the relevant pivotal tracker stories?
[#167912173](https://www.pivotaltracker.com/story/show/167912173)
#### Screenshots
![Screenshot (68)](https://user-images.githubusercontent.com/46356108/63085418-c7623580-bf4d-11e9-8b3d-e299a80db635.png)
![Screenshot (69)](https://user-images.githubusercontent.com/46356108/63085419-c7623580-bf4d-11e9-8cb3-63a9b2c09ef0.png)
![Screenshot (65)](https://user-images.githubusercontent.com/46356108/63085421-c7facc00-bf4d-11e9-940a-e17305b17a60.png)
![Screenshot (66)](https://user-images.githubusercontent.com/46356108/63085422-c7facc00-bf4d-11e9-8151-8e8101fda2dc.png)
![Screenshot (67)](https://user-images.githubusercontent.com/46356108/63085424-c7facc00-bf4d-11e9-9a16-edc589e80bdd.png)

amkayondo committed:
ft(API): User can delete a booking
- Get booking by Id
- increase the number of seats available on trip
- add validation for invalid
[finishes #167912173]

Kasulejoseph commented on server/models/bookings.js:
```diff
@@ -30,12 +39,14 @@ export default class Book {
 
-  async checkIfTripExists(trip_id) {
-    const result = await newTrip.getTripById(trip_id);
+  async checkIfTripExistsById(tripId) {
```
I think `isTripIdExixts` would be a better method name here

Kasulejoseph commented on server/models/bookings.js:
```diff
@@ -20,7 +20,16 @@ export default class Book {
+    return result;
+  }
+
+  async increaseNumberOfSeats(tripId, seatsAvailable, newSeats) {
```
`decreaseNumberOfSeats` and `increaseNumberOfSeats` can be a single method try refactoring..

Kasulejoseph commented on server/models/bookings.js:
```diff
@@ -20,7 +20,16 @@ export default class Book {
+    const newData = seatsAvailable + newSeats;
     console.log(newData);
+    console.log(seatsAvailable);
+    console.log(newSeats);
```
Remove the debuger statements as well

Kasulejoseph commented on server/models/bookings.js:
```diff
@@ -30,12 +39,14 @@ export default class Book {
-  async increaseNumberOfSeats(findTrip, seats) {
-    return parseInt(findTrip + seats);
+
+  async checkIfTripExists(bus_license_number, trip_date) {
```
`isTripExists` would be better

Kasulejoseph commented on server/models/bookings.js:
```diff
@@ -47,12 +58,18 @@ export default class Book {
+  }
+
   async checkBookingUser(userid) {
     // return this.getAtllbookings.find(x => x.userId === userid);
```
What's the purpose of this method? If it doesn't do anything please remove it

amkayondo replied:
I am going to use it when am getting user bookings

Kasulejoseph commented:
In the response, you're having a typo `massage` instead of `message` please correct it

amkayondo committed:
chore(API): implement feedback
- decreaseNumberOfSeats and increaseNumberOfSeats method refactoring
- change checkIfTripExistsById to isTripIdExixts
[finishes #167912173]

amkayondo committed:
chore(API): implement feedback
- change checkIfTripExists to isTripExists
[finishes #167912173]

Kasulejoseph commented on server/controllers/bookings/getAllBookings.js:
```diff
@@ -2,22 +2,23 @@ import jwt from 'jsonwebtoken';
+    return resPonse.errorMessage(res, 200, 'You have made no bookings yet');
+  } resPonse.successData(res, 200, forUser);
+  // } catch (error) {
+  //   resPonse.errorMessage(res, 500, error.message);
```
Why are you commenting this out, remove it entirely if you think it doesn't add any value.

amkayondo committed:
chore(API): implement feedback
- change massage to message
